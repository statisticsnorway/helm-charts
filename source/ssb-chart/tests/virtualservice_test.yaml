suite: test virtualservice
templates:
  - virtualservice.yaml
tests:

  - it: should render nothing if not exposed
    asserts:
      - hasDocuments:
          count: 0

  - it: should render one document if exposed
    values:
      - ./values/default-values.yaml
    set:
      access:
        ingress:
          external:
            gateways:
              - type: public
    asserts:
      - hasDocuments:
          count: 1

  - it: should render nothing if exposed is a string other than "public"
    values:
      - ./values/default-values.yaml
    set:
      access:
        ingress:
          external:
            gateways:
              - type: other
    asserts:
      - hasDocuments:
          count: 0

  - it: should generate manifest of type VirtualService
    values:
      - ./values/virtualservice-values.yaml
    set:
      access:
        ingress:
          external:
            gateways:
              - type: public
    asserts:
      - isKind:
          of: VirtualService

  - it: should generate gateway
    values:
      - ./values/virtualservice-values.yaml
    set:
      access:
        ingress:
          external:
            gateways:
              - type: public
    asserts:
      - contains:
          path: spec.gateways
          content:
            istio-system/test-cluster-gateway

  - it: should generate host and subdomains
    values:
      - ./values/virtualservice-values.yaml
    set:
      access:
        ingress:
          external:
            gateways:
              - type: public
      subdomains:
        - testdomain1
        - testdomain2
    release:
      namespace: unittest
    asserts:
      - equal:
          path: spec.hosts
          value:
            - unittest-app.testdomain1.ssb.no
            - unittest-app.testdomain2.ssb.no
            - unittest-app.test-cluster.ssb.no

  - it: should generate host
    values:
      - ./values/virtualservice-values.yaml
    set:
      access:
        ingress:
          external:
            gateways:
              - type: public
    release:
      namespace: unittest
    asserts:
      - equal:
          path: spec.hosts
          value:
            - unittest-app.test-cluster.ssb.no

  - it: should generate urlhost
    values:
      - ./values/virtualservice-values.yaml
    set:
      access:
        ingress:
          external:
            gateways:
              - type: public
      urlhosts:
        - testurl1
        - testurl2
    release:
      namespace: unittest
    asserts:
      - equal:
          path: spec.hosts
          value:
            - testurl1.ssb.no
            - testurl2.ssb.no
            - unittest-app.test-cluster.ssb.no

  - it: should generate urlhosts, subdomains
    values:
      - ./values/virtualservice-values.yaml
    set:
      access:
        ingress:
          external:
            gateways:
              - type: public
      urlhosts:
        - testurl1
        - testurl2
      subdomains:
        - testdomain1
        - testdomain2
    release:
      namespace: unittest
    asserts:
      - equal:
          path: spec.hosts
          value:
            - testurl1.testdomain1.ssb.no
            - testurl2.testdomain1.ssb.no
            - testurl1.testdomain2.ssb.no
            - testurl2.testdomain2.ssb.no
            - unittest-app.test-cluster.ssb.no

  - it: should generate default routes and extra routes
    values:
      - ./values/virtualservice-values.yaml
    set:
      access:
        ingress:
          external:
            gateways:
              - type: public
    release:
      namespace: unittest
    asserts:
      - matchSnapshot:
          path: spec.http
