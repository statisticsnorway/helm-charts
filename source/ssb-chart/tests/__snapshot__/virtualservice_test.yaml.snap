should generate default routes and extra routes:
  1: |
    - match:
        - uri:
            prefix: /be/unittest-app/auth
      route:
        - destination:
            host: unittest-app.unittest.svc.cluster.local
            port:
              number: 8080
    - match:
        - uri:
            prefix: /oauth2
      route:
        - destination:
            host: unittest-app.unittest.svc.cluster.local
            port:
              number: 8080
    - match:
        - uri:
            prefix: /auth
      route:
        - destination:
            host: unittest-app.unittest.svc.cluster.local
            port:
              number: 8080
    - match:
        - uri:
            prefix: /metrics
      rewrite:
        uri: /
      route:
        - destination:
            host: unittest-app.unittest.svc.cluster.local
            port:
              number: 80
    - route:
        - destination:
            host: unittest-app.unittest.svc.cluster.local
            port:
              number: 80
